src/main/java/Parser.java

```diff
@@ -0,0 +1,108 @@
+import java.time.LocalDate;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * Encapsulates parser into a class. The class supports parsing the command messages
+ * from the user into different commands.
+ */
+public class Parser {
+    public static Command parse(String commandMessage) throws DukeException {
+        Command command;
+        if (commandMessage.equals("list")) {
+            command = new Command("list");
+        } else if (commandMessage.contains("done")) {
+            int order = parseInt(commandMessage.substring(commandMessage.length() - 1));
+            command = new Command("done", order);
+        } else if (commandMessage.contains("delete")) {
+            int order = parseInt(commandMessage.substring(commandMessage.length() - 1));
+            command = new Command("delete", order);
+        } else if (commandMessage.contains("find")) {
+            command = new Command("find", commandMessage.substring(commandMessage.indexOf(' ') + 1));
+        } else if (commandMessage.equals("bye")) {
+            command = new Command("bye");
```

Comment:
May I know why you represent all commands with one class instead of one class for one command?

Reply:
I think its because it is not a requirement for the More-OOP tag but a good to have!